src/acceleration/impl/Preconditioner.hpp

```diff
@@ -58,16 +58,20 @@ class Preconditioner {
   {
     PRECICE_TRACE();
     if (transpose) {
-      PRECICE_ASSERT(M.cols() == (int) _weights.size(), M.cols(), _weights.size());
-      for (int i = 0; i < M.cols(); i++) {
+      PRECICE_DEBUG_IF((int) _weights.size() != M.cols(), "The number of columns of the matrix {} and weights size {} mismatched.", M.rows(), _weights.size());
```

Comment:
```suggestion
      PRECICE_DEBUG_IF(static_cast<Eigen::Index>(_weights.size()) != M.cols(), "The number of columns of the matrix {} and weights size {} mismatched.", M.rows(), _weights.size());
```
We should avoid the C-style casting. I think clang-tidy should complain as well